src/Chapter3.hs

```diff
@@ -374,6 +389,46 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Fighter = MkFighter
+    { fighterHealthPts    :: Int,
+      fighterAttackPts    :: Int,
+      fighterGold         :: Int
+    } deriving (Show)
+
+knight :: Fighter
+knight = MkFighter
+    { fighterHealthPts = 10,
+      fighterAttackPts = 2,
+      fighterGold = 5
+    }
+
+monster :: Fighter
+monster = MkFighter
+    { fighterHealthPts = 15,
+      fighterAttackPts = 1,
+      fighterGold = 11
+    }
+
+
+fight :: Fighter -> Fighter -> Int
```

Comment:
I see that you use the same type for `Monster` and `Knight`. This is a nice catch, because these two entities have the same structure and characteristics.
The only issue with reusage of the same data type for Monster and Knight in here is that you can easily mix them when calling the function:

```haskell
> fight knight monster
> fight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two types for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Entity
newtype Knight = Knight Entity
```